### PR TITLE
Boltz service

### PR DESCRIPTION
**Pull request by ImmanuelSegol**

### CODE REVIEW

File: docker/boltz-backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+FROM node:8.11.3
```
Comment (michael1011) [inline docker/boltz-backend/Dockerfile]: I would go for the tag `lts-alpine` of the `node` image.

File: docker/boltz-backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		libc6-dev \
+		make \
+		pkg-config \
+	&& rm -rf /var/lib/apt/lists/*
+
+WORKDIR /opt
+
+# Install dependencies
+RUN apt-get update && apt-get install -y rsync
+RUN npm install -g gulp nodemon typescript --quiet
```
Comment (michael1011) [inline docker/boltz-backend/Dockerfile]: Why do we need to install this dependencies globally?

File: docker/boltz-backend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
+FROM node:8.11.3
+
+
+# Install general dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
```
Comment (michael1011) [inline docker/boltz-backend/Dockerfile]: Would splitting up the Docker file into a build stage and a final stage to reduce the size of the image work here?

File: docker/boltz-service/boltz.conf [hunk 4]
```diff
@@ -0,0 +1,32 @@
+[[currencies]]
+  symbol = "BTC"
+  network = "bitcoinSimnet"
```
Comment (michael1011) [inline docker/boltz-service/boltz.conf]: We are not using the simnet but regtest network

File: docker/boltz-service/boltz.conf [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  [currencies.lnd]
+    host = "127.0.0.1"
+    port = 10009
+    certpath = "/opt/boltz-backend/docker/regtest/data/lnd/certificates/tls.cert"
+    macaroonpath = "/opt/boltz-backend/docker/regtest/data/lnd/macaroons/admin.macaroon"
+
+[[currencies]]
+  symbol = "LTC"
+  network = "litecoinSimnet"
```
Comment (michael1011) [inline docker/boltz-service/boltz.conf]: Same here: regtest not simnet

File: docker/boltz-service/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+		gcc \
+		libc6-compat \
+		make \
+		pkgconfig \
+		rsync \
+		python \
+		--no-cache git \
+	&& rm -rf /var/lib/apt/lists/*
+
+RUN npm install gulp nodemon typescript --quiet
```
Comment (michael1011) [inline docker/boltz-service/Dockerfile]: Why is installing these dependencies necessary?

File: docker/build-images.sh [hunk 7]
```diff
@@ -18,3 +18,6 @@ docker build -t boltz/lnd -f lnd/Dockerfile .
 
 print "regtest image"
 docker build -t boltz/regtest -f regtest/Dockerfile .
+
+print "boltz image"
+docker build -t boltz/service -f boltz-service/Dockerfile .
```
Comment (michael1011) [inline docker/build-images.sh]: What about calling the image `boltz/backend`?

File: docker/boltz-service/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+# Expose RPC port
+EXPOSE 9000
+
+# Compile TypeScript
+RUN npm run compile
+
+# Copty config file
+WORKDIR /root/.boltz
+COPY boltz.conf boltz.conf
```
Comment (michael1011) [inline docker/boltz-service/Dockerfile]: The build fails for me with 

```
COPY failed: stat /var/lib/docker/tmp/docker-builder175635307/boltz.conf: no such file or directory
```

File: docker/boltz-service/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+WORKDIR /opt/boltz-backend
+COPY --from=builder /opt/boltz-backend .
+
+# Expose RPC port
+EXPOSE 9000
+
+# Compile TypeScript
+RUN npm run compile
+
+# Copty config file
```
Comment (michael1011) [inline docker/boltz-service/Dockerfile]:
```suggestion
# Copy config file
```

File: docker/boltz-service/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+RUN apk update && apk add rsync
+# Copy boltz-backend from builder image
+WORKDIR /opt/boltz-backend
+COPY --from=builder /opt/boltz-backend .
+
+# Expose RPC port
+EXPOSE 9000
+
+# Compile TypeScript
+RUN npm run compile
```
Comment (michael1011) [inline docker/boltz-service/Dockerfile]: What about compiling the code in the build stage and just copying over the compiled JavaScript to the final image?

File: docker/boltz-service/boltz.conf (hunk 4 above)
Comment (ImmanuelSegol) [reply docker/boltz-service/boltz.conf]: Oops, thanks for pointing that out.

File: docker/boltz-service/Dockerfile (hunk 6 above)
Comment (ImmanuelSegol) [reply docker/boltz-service/Dockerfile]: Correct my mistake they are not necessary.

File: docker/boltz-service/Dockerfile (hunk 8 above)
Comment (ImmanuelSegol) [reply docker/boltz-service/Dockerfile]: It works for me both on Linux and windows.

`Docker version 18.09.3, build 774a1f4`

File: docker/boltz/boltz-backend/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,40 @@
+FROM node:lts-alpine as builder
+# Install general dependencies
```
Comment (michael1011) [inline docker/boltz/boltz-backend/Dockerfile]:
```suggestion

# Install dependencies
```

File: docker/boltz/boltz-backend/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+# Clone boltz-backend
+WORKDIR /opt
+RUN git clone https://github.com/BoltzExchange/boltz-backend
+WORKDIR /opt/boltz-backend
+RUN npm install --quiet
+RUN npm run compile
+
+FROM node:lts-alpine
+RUN apk update && apk add rsync
+# Copy boltz-backend from builder image
```
Comment (michael1011) [inline docker/boltz/boltz-backend/Dockerfile]:
```suggestion

# Copy boltz-backend from builder image
```

File: docker/boltz/boltz-backend/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+# Clone boltz-backend
+WORKDIR /opt
+RUN git clone https://github.com/BoltzExchange/boltz-backend
+WORKDIR /opt/boltz-backend
+RUN npm install --quiet
+RUN npm run compile
+
+FROM node:lts-alpine
+RUN apk update && apk add rsync
```
Comment (michael1011) [inline docker/boltz/boltz-backend/Dockerfile]: `rsync` is just needed for `npm run compile` and therefore not needed in the final image

File: docker/boltz/boltz-backend/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+# Copy boltz-backend from builder image
+WORKDIR /opt/boltz-backend
+COPY --from=builder /opt/boltz-backend .
+
+# Expose RPC port
+EXPOSE 9000
+
+# Copy config file
```
Comment (michael1011) [inline docker/boltz/boltz-backend/Dockerfile]:
```suggestion
# Copy config file and TLS certificate
```

File: docker/boltz/boltz-backend/Dockerfile [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+WORKDIR /root/.boltz
+COPY boltz-backend/certificates/tls.cert tls.cert
+COPY boltz-backend/certificates/tls.key tls.key 
+COPY boltz-backend/boltz.conf boltz.conf 
+
+WORKDIR /opt/boltz-backend
+
+# Start boltz service
+ENTRYPOINT ["npm","start"]
```
Comment (michael1011) [inline docker/boltz/boltz-backend/Dockerfile]:
```suggestion
ENTRYPOINT ["npm", "start"]
```

File: docker/boltz/build-images.sh [hunk 16]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+function print () {
+  echo -e "\e[32mBuilding $1\e[0m"
+}
+
+print "boltz-backend image"
+docker build -t boltz/backend -f boltz-backend/Dockerfile .
```
Comment (michael1011) [inline docker/boltz/build-images.sh]: This should be moved to the already existing build script

**Comment by armurbalda** [comment]
stale, closing

**Pull request closed without merging**